src/gt4py/storage/storage.py

```diff
@@ -14,604 +14,1112 @@
-
-    def __deepcopy__(self, memo={}):
-        return self.copy()
-
-    def copy(self):
-        res = empty(
-            shape=self.shape,
-            dtype=self.dtype,
-            backend=self.backend,
-            default_origin=self.default_origin,
-            mask=self.mask,
-            managed_memory=not isinstance(self, ExplicitlySyncedGPUStorage),
-        )
-        res.is_stencil_view = self.is_stencil_view
-        return res
-
-    def __array_finalize__(self, obj):
-        if obj is None:
-            # constructor called previously
-            return
-        else:
-            if self.base is None:
-                # case np.array
-                raise RuntimeError(
-                    "Copying storages is only possible through Storage.copy() or deepcopy."
-                )
-            else:
-                if not isinstance(obj, Storage) and not isinstance(obj, _ViewableNdarray):
-                    raise RuntimeError(
-                        "Meta information can not be inferred when creating Storage views from other classes than Storage."
-                    )
-                self.__dict__ = {**obj.__dict__, **self.__dict__}
-                self.is_stencil_view = False
-                if not hasattr(obj, "default_origin"):
-                    self.is_stencil_view = True
-                elif self._is_consistent(obj):
-                    self.is_stencil_view = obj.is_stencil_view
-                self._finalize_view(obj)
-
-    def _is_consistent(self, obj):
-        if not self.shape == obj.shape:
-            return False
-        # check strides
-        stride = 0
-        if len(self.strides) < len(self.layout_map):
-            return False
-        for dim in reversed(np.argsort(self.layout_map)):
-            if self.strides[dim] < stride:
-                return False
-            stride = self.strides[dim]
-
-        # check alignment
-        if (
-            self.ctypes.data + np.sum([o * s for o, s in zip(self.default_origin, self.strides)])
-        ) % gt_backend.from_name(self.backend).storage_info["alignment"]:
-            return False
-
-        return True
-
-    def _finalize_view(self, obj):
-        pass
-
-    def synchronize(self):
-        pass
-
-    def host_to_device(self, force=False):
-        pass
-
-    def device_to_host(self, force=False):
-        pass
-
-    def __iconcat__(self, other):
-        raise NotImplementedError("Concatenation of Storages is not supported")
-
-
-class GPUStorage(Storage):
-    @classmethod
-    def _construct(cls, backend, dtype, default_origin, shape, alignment, layout_map):
-
-        (raw_buffer, field) = storage_utils.allocate_gpu(
-            default_origin, shape, layout_map, dtype, alignment * dtype.itemsize
-        )
-        obj = field.view(_ViewableNdarray)
-        obj = obj.view(GPUStorage)
-        obj._raw_buffer = raw_buffer
-        obj.default_origin = default_origin
-        return obj
+    def axes(self):
```

Comment:
This attribute only occurs once in the tests within an integration test, and could use some unit tests.